File: java/javafind/src/test/java/javafind/FileResultTest.java
package javafind;

import org.junit.jupiter.api.Test;

import java.nio.file.Paths;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class FileResultTest {

    public FileResultTest() {}

    @Test
    public final void testFileResultAbsPath() {
        var dir = "~/src/xfind/java/javafind/src/main/java/javafind";
        var fileName = "FileResult.java";
        var path = Paths.get(dir + "/" + fileName);
        var fileResult = new FileResult(path, FileType.CODE);
        var expectedFilePath = dir + "/" + fileName;
        assertEquals(expectedFilePath, fileResult.toString());
    }

    @Test
    public final void testFileResultRelPath1() {
        var dir = ".";
        var fileName = "FileResult.java";
        var path = Paths.get(dir + "/" + fileName);
        var fileResult = new FileResult(path, FileType.CODE);
        var expectedFilePath = dir + "/" + fileName;
        assertEquals(expectedFilePath, fileResult.toString());
    }

    @Test
    public final void testFileResultRelPath2() {
        var dir = "..";
        var fileName = "FileResult.java";
        var path = Paths.get(dir + "/" + fileName);
        var fileResult = new FileResult(path, FileType.CODE);
        var expectedFilePath = dir + "/" + fileName;
        assertEquals(expectedFilePath, fileResult.toString());
    }
}

File: kotlin/ktfind/src/test/kotlin/ktfind/FileResultTest.kt
package ktfind

import java.nio.file.Paths
import kotlin.test.Test
import kotlin.test.assertEquals

class FileResultTest {

    @Test
    fun testFileResultAbsPath() {
        val path = "/Users/cary/src/xfind/kotlin/ktfind/src/main/kotlin/ktfind/FileResult.kt"
        val fileResult = FileResult(Paths.get(path), FileType.CODE)
        assertEquals(path, fileResult.toString())
    }

    @Test
    fun testFileResultTildePath() {
        val path = "~/src/xfind/kotlin/ktfind/src/main/kotlin/ktfind/FileResult.kt"
        val fileResult = FileResult(Paths.get(path), FileType.CODE)
        assertEquals(path, fileResult.toString())
    }

    @Test
    fun testFileResultRelPath1() {
        val path = "./FileResult.kt"
        val fileResult = FileResult(Paths.get(path), FileType.CODE)
        assertEquals(path, fileResult.toString())
    }

    @Test
    fun testFileResultRelPath2() {
        val path = "../FileResult.kt"
        val fileResult = FileResult(Paths.get(path), FileType.CODE)
        assertEquals(path, fileResult.toString())
    }
}
